crates/nu-cli/src/cli.rs

```diff
@@ -323,6 +323,9 @@ pub async fn run_vec_of_pipelines(
 #[cfg(feature = "rustyline-support")]
 fn convert_rustyline_result_to_string(input: Result<String, ReadlineError>) -> LineResult {
     match input {
+        Ok(s) if s.contains("history -c") || s.contains("history --clear") => {
```

Comment:
Searching for this specifically seems like it might be problematic.

If the user made an input that was `echo "history -c"` would that clear their history?

Reply:
That's a good point, I originally did a `contains` in case it was part of a piped command, but it might be problematic. 

What about doing a strict equals match? Or is there some other approach you have in mind?

Reply:
Updated to be a strict equals match